### PR TITLE
add specfile to build srpm

### PR DESCRIPTION
**Pull request by miyunari**
Follow up to https://github.com/os-observability/redhat-opentelemetry-collector/pull/40

### CODE REVIEW

File: collector.spec [hunk 1]
```diff
@@ -0,0 +1,54 @@
+%global goipath         github.com/os-observability/redhat-opentelemetry-collector
+
+Version:                0.92.0
```
Comment (frzifus) [inline collector.spec:3]: Can we use the verison defined in the `Makefile` here?

File: collector.spec [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+ExcludeArch:            %{ix86} s390 ppc ppc64 aarch64
+
+%gometa
+
+%global common_description %{expand:
+TODO}
```
Comment (frzifus) [inline collector.spec:9]:
```suggestion
Collector with the supported components for a Red Hat build of OpenTelemetry product}
```

File: collector.spec [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+
+%global golicenses    LICENSE
+%global godocs        README.md
+
+Name:           redhat-opentelemetry-collector
```
Comment (frzifus) [inline collector.spec:14]: Can we use the name `defined` in the Makefile here?

File: collector.spec [hunk 4]
```diff
@@ -0,0 +1,54 @@
...
+%description
+%{common_description}
+
+%gopkg
+
+%prep
+%goprep
+
+%build
+make build
```
Comment (frzifus) [inline collector.spec:35]: Does `make` add some additonal `BuildRequires`?

File: collector.spec [hunk 5]
```diff
@@ -0,0 +1,54 @@
...
+
+%global common_description %{expand:
+TODO}
+
+%global golicenses    LICENSE
+%global godocs        README.md
+
+Name:           redhat-opentelemetry-collector
+Release:        1%{?dist}
+Summary:        TODO
```
Comment (frzifus) [inline collector.spec:16]:
````suggestion
Summary:        Red Hat build of OpenTelemetry
```

File: collector.spec [hunk 6]
```diff
@@ -0,0 +1,55 @@
...
+%files
+%license %{golicenses}
+%doc %{godocs}
+%{_bindir}/*
+
+%gopkgfiles
+
+%changelog
+* Thu Feb 1 21:59:10 CET 2024 Nina Olear <nolear@redhat.com> - 0.93.4
+- First package for Copr
```
Comment (poncovka) [inline collector.spec:55]: Please, add an extra line here.

File: collector.spec (hunk 1 above)
Comment (poncovka) [reply collector.spec:3]: Yeah, you can use `sed`. There is usually also some automation for bumping the version. For example:

- https://github.com/pykickstart/pykickstart/blob/master/pykickstart.spec.in#L7
- https://github.com/pykickstart/pykickstart/blob/master/Makefile#L1
- https://github.com/pykickstart/pykickstart/blob/master/Makefile#L117-L128

However, I would explore what packit can do first (https://packit.dev/). I am not familiar with latests trends and I remember some discussion about managing the spec file. However, that might have beeen the downstream spec file.

File: collector.spec (hunk 3 above)
Comment (poncovka) [reply collector.spec:14]: Same as above.

File: collector.spec (hunk 4 above)
Comment (poncovka) [reply collector.spec:36]: I guess so? https://github.com/rhinstaller/anaconda/blob/master/anaconda.spec.in#L61

**Comment by miyunari** [comment]
Fedpkg is not installed on the github ci worker. Therefore I removed the make rule for now. :sweat_smile:

**Comment by poncovka** [comment]
> Fedpkg is not installed on the github ci worker. Therefore I removed the make rule for now. 😅

I don't think you need to test a build of the SRPM, but it would be good to eventually build the RPM, install it and verify that the binary file exists at the expected places (or call `--help` or something, if it is supported, it is a great smoke test). I would suggest to use packit for that. For example, check: https://github.com/rhinstaller/dasbus/blob/master/.packit.yaml

File: collector.spec [hunk 7]
```diff
@@ -0,0 +1,59 @@
...
+BuildRequires: git
+BuildRequires: make
+
+%description
+%{common_description}
+
+%gopkg
+
+%prep
+%goprep
```
Comment (major) [inline collector.spec:37]: Don't we need a `%goprep -k` for bundling?

File: collector.spec [hunk 8]
```diff
@@ -0,0 +1,59 @@
...
+Source1:        %{name}-deps-%{version}.tar.gz
+
+BuildRequires: go-toolset-1.20
+BuildRequires: git
+BuildRequires: make
+
+%description
+%{common_description}
+
+%gopkg
```
Comment (major) [inline collector.spec:34]: If we're planning to bundle and only package the final binary, `%gopkg` needs to be removed.

File: collector.spec [hunk 9]
```diff
@@ -0,0 +1,59 @@
...
+
+%prep
+%goprep
+
+%build
+make build
+
+%install
+%gopkginstall
```
Comment (major) [inline collector.spec:43]: Remove `%gopkginstall` if we are bundling.

File: collector.spec [hunk 10]
```diff
@@ -0,0 +1,59 @@
...
+
+%check
+%gocheck
+
+%files
+%license %{golicenses}
+%doc %{godocs}
+%{_bindir}/*
+
+%gopkgfiles
```
Comment (major) [inline collector.spec:55]: We need `%gopkgfiles` removed if we're bundling.

**Comment by miyunari** [comment]
Thanks @major, I have adopted your suggested changes 😄 🚀

**Comment by frzifus** [comment]
cc @sallyom

File: collector.spec [hunk 11]
```diff
@@ -0,0 +1,54 @@
+%global goipath         github.com/os-observability/redhat-opentelemetry-collector
```
[the remark this answers is not in the document]
Comment (frzifus) [reply collector.spec]: I added a job to verify the srpm.

**Review by frzifus — APPROVED** [review]

**Pull request merged**